src/br/com/alura/kotlin/collection/list/ListaDeLivros.kt

```diff
@@ -0,0 +1,79 @@
+package br.com.alura.kotlin.collection.list
+
+fun main() {
+    val meusLivros: MutableList<Livro> = mutableListOf(
+        Livro(nome = "Grande Sertão: Veredas", autor = "João Guimarães Rosa", anoPublicacao = 1956),
+        Livro(nome = "Minha vida de menina", autor = "Helena Morley", anoPublicacao = 1942, editora = "Editora A"),
+        Livro(nome = "Memórias Póstumas de Brás Cubas", autor = "Machado de Assis", anoPublicacao = 1881),
+        Livro(nome = "Iracema", autor = "José de Alencar", anoPublicacao = 1865, editora = "Editora B")
+    )
+    println(meusLivros)
+
+    meusLivros.add(Livro(nome = "Sagarana", autor = "João Guimarães Rosa", anoPublicacao = 1946))
+
+    println()
+    println(meusLivros.joinToString(separator = "\n") { " - ${it.nome} de ${it.autor}" })
+
+    val meusLivrosComFiltro = meusLivros
+        .filter { it.autor.startsWith("J") }
+        .sortedBy { it.anoPublicacao }
+        .map { it.nome }
+
+    println()
+    println(meusLivrosComFiltro)
+
+    val meusLivrosComNulls = mutableListOf<Livro?>(null, null)
+    meusLivrosComNulls.addAll(meusLivros)
+
+    println()
+    println(meusLivrosComNulls.joinToString(separator = "\n") { " - ${it?.nome} de ${it?.autor}" })
+
+    println()
+    println(meusLivrosComNulls.filter { it != null }.joinToString(separator = "\n") { " - ${it!!.nome} de ${it.autor}" })
```

Comment:
Lembrar logo de cara que !! não é recomendado, acho que nem vale a pena deixar o código, pode converter para algo mais seguro.

Reply:
A ideia era justamente mostrar a diferença entre usar `filter { it != null }` e `filterNotNull()` e explicar um pouco sobre smart cast. Por isso que eu mantive o `!!`.